packages/core/src/modules/credentials/services/CredentialService.ts

```diff
@@ -297,6 +315,24 @@ export class CredentialService {
       }),
     })
 
+    // Check if credential preview attributes match the schema attributes
+    const schema = await this.ledgerService.getSchema(credOffer.schema_id)
+    const credAttributes = credentialTemplate.preview.attributes.map((a) => a.name)
+    const schemaAttributes = schema.attrNames
+
+    const difference: string[] = []
+    credAttributes.forEach((credAtrr) => {
+      if (schemaAttributes.indexOf(credAtrr) === -1) {
+        difference.push(credAtrr)
+      }
+    })
```

Comment:
Looking at this again this is not where we landed yesterday right? I think we took the example of "symmetric difference". Currently if you have a property missing in the credAttributes array, it won't throw an error.

https://stackoverflow.com/questions/1187518/how-to-get-the-difference-between-two-arrays-in-javascript

This will find _any_ differences between the two arrays:

```suggestion
    const difference = credAttributes
                 .filter(x => !schemaAttributes.includes(x))
                 .concat(schemaAttributes.filter(x => !credAttributes.includes(x)));
```